internal/bminventory/inventory.go

```diff
@@ -1973,8 +1976,9 @@ func (b *bareMetalInventory) updateClusterData(_ context.Context, cluster *commo
 func (b *bareMetalInventory) updateNetworks(db *gorm.DB, params installer.V2UpdateClusterParams, updates map[string]interface{},
 	cluster *common.Cluster, userManagedNetworking, vipDhcpAllocation bool) error {
 	var err error
+	var updated bool
 
-	if params.ClusterUpdateParams.ClusterNetworks != nil {
+	if params.ClusterUpdateParams.ClusterNetworks != nil && !network.AreClusterNetworksIdentical(params.ClusterUpdateParams.ClusterNetworks, cluster.ClusterNetworks) {
```

Comment:
Would it actually be better to do some kind of set difference here so that when the networks are updated we only run the verification on the ones that are new?

Reply:
Isn't that what we do here?